jdk/src/share/classes/sun/security/pkcs11/P11Key.java

```diff
@@ -366,7 +372,20 @@ static PrivateKey privateKey(Session session, long keyID, String algorithm,
             new CK_ATTRIBUTE(CKA_SENSITIVE),
             new CK_ATTRIBUTE(CKA_EXTRACTABLE),
         });
-        if (attributes[1].getBoolean() || (attributes[2].getBoolean() == false)) {
+        if (attributes[0].getBoolean() || attributes[1].getBoolean() || (attributes[2].getBoolean() == false)) {
+            if ((SunPKCS11.mysunpkcs11 != null) && "RSA".equals(algorithm)) {
+                try {
+                    byte[] key = SunPKCS11.mysunpkcs11.exportKey(session.id(), attributes, keyID);
+                    RSAPrivateKey rsaPrivKey = RSAPrivateCrtKeyImpl.newKey(key);
+                    if (rsaPrivKey instanceof RSAPrivateCrtKeyImpl) {
+                        return new P11RSAPrivateKeyFIPS(session, keyID, algorithm, keyLength, attributes, (RSAPrivateCrtKeyImpl)rsaPrivKey);
+                    } else {
+                        return new P11RSAPrivateNonCRTKeyFIPS(session, keyID, algorithm, keyLength, attributes, rsaPrivKey);
+                    }
+                } catch (PKCS11Exception | InvalidKeyException e) {
+                    // Attempt failed, create a P11PrivateKey object.
+                }
+            }
```

Comment:
The extra attribute check should only be for the FIPS case:
```
        if ((SunPKCS11.mysunpkcs11 != null) && "RSA".equals(algorithm)) {
            if (attributes[0].getBoolean() || attributes[1].getBoolean() || (attributes[2].getBoolean() == false)) {
                try {
                    byte[] key = SunPKCS11.mysunpkcs11.exportKey(session.id(), attributes, keyID);
                    RSAPrivateKey rsaPrivKey = RSAPrivateCrtKeyImpl.newKey(key);
                    if (rsaPrivKey instanceof RSAPrivateCrtKeyImpl) {
                        return new P11RSAPrivateKeyFIPS(session, keyID, algorithm, keyLength, attributes, (RSAPrivateCrtKeyImpl)rsaPrivKey);
                    } else {
                        return new P11RSAPrivateNonCRTKeyFIPS(session, keyID, algorithm, keyLength, attributes, rsaPrivKey);
                    }
                } catch (PKCS11Exception | InvalidKeyException e) {
                    // Attempt failed, create a P11PrivateKey object.
                }
            }
        }
        if (attributes[1].getBoolean() || (attributes[2].getBoolean() == false)) {
```

Reply:
updated